src/node_report.cc

```diff
@@ -259,6 +267,39 @@ static void WriteNodeReport(Isolate* isolate,
 
   writer.json_arrayend();
 
+  writer.json_arraystart("workers");
+  if (env != nullptr) {
+    Mutex workers_mutex;
+    ConditionVariable notify;
+    std::vector<std::string> worker_infos;
+    size_t expected_results = 0;
+
+    env->ForEachWorker([&](Worker* w) {
+      expected_results += w->RequestInterrupt([&](Environment* env) {
+        std::ostringstream os;
+
+        GetNodeReport(env->isolate(),
+                      env,
+                      "Worker thread subreport",
+                      trigger,
+                      Local<String>(),
+                      os);
+
+        Mutex::ScopedLock lock(workers_mutex);
+        worker_infos.emplace_back(os.str());
+        notify.Signal(lock);
+      });
+    });
+
+    Mutex::ScopedLock lock(workers_mutex);
+    worker_infos.reserve(expected_results);
+    while (worker_infos.size() < expected_results)
+      notify.Wait(lock);
+    for (const std::string& worker_info : worker_infos)
+      writer.json_element(JSONWriter::ForeignJSON { worker_info });
+  }
+  writer.json_arrayend();
+
```

Comment:
I am getting an empty `workers: []` block everytime, apart from a valid block, did you observe that?

Reply:
I don’t know if this is what you’re referring to, but the report only includes Workers which have an `Environment` of their own. That does not happen synchronously, hence the `await once(w, 'online');` lines in the test; without that, the `workers` array would be empty.

Reply:
got it, understood.